### PR TITLE
Add pwd to docker container

### PR DESCRIPTION
**Pull request by klarkc**
Allows the use of current dir `.gtt.yml` inside the docker container.

### CODE REVIEW

**Review by cgdobre — CHANGES_REQUESTED** [review]
Thanks for the contribution. What do you think about this idea instead to get the same result? 
`docker run --rm -it -v ~:/root -v $(pwd)/.gtt.yml:/.gtt.yml kriskbx/gitlab-time-tracker'`
This way it wouldn't be necessary to change anything else than the documentation.

**Comment by klarkc** [comment]
> What do you think about this idea instead to get the same result?

Almost, if `.gtt.yml` file does not exists on host it will not throw this error: 

```
$ docker run --rm -it -v ~:/root -v $(pwd)/.gtt.yml:/.gtt.yml kriskbx/gitlab-time-tracker report
Error parsing configuration: "/.gtt.yml"
```

Because, when a given path does not exists, docker creates a folder for the volume:

```
$ ls -la .gtt.yml/
total 8
drwxr-xr-x  2 root   root   4096 jun  2 12:51 .
drwxr-xr-x 18 klarkc klarkc 4096 jun  2 12:51 ..
```

As `.gtt.yml` file is optional, I think this way is more painless useful for use with an alias or inside a script.

File: documentation.md [hunk 1]
```diff
@@ -70,12 +70,13 @@ you can use the official [Docker image](https://hub.docker.com/r/kriskbx/gitlab-
...
        --rm -it \
        -v ~:/root \
+       -v $(pwd):/pwd \
        kriskbx/gitlab-time-tracker \
        --help
 ```
 
 `--rm` removes the container after running, `-it` makes it interactive, `-v ~:/root` mounts your home directory to the
-home directory inside the container. If you want to store the config in another place, mount another directory: 
+home directory inside the container, `-v $(pwd):/pwd` mounts current directory inside the container to gtt be able to read local config. If you want to store the config in another place, mount another directory:
```
Comment (juanluisbaptiste) [inline documentation.md:79]: The correct env variables is PWD, env vars are case sensitive so this will not work.

**Comment by juanluisbaptiste** [comment]
Hi,

This PR does not fix the issue, first, env variable names are case sensitive and the correct env variable name for the current path value is `${PWD}`, `${pwd}` does return an empty value. Second, the volume mount for the /root directory using "~" is still present so the `docker run` command will still fail. Third, you don't need to mount the complete user home directory, that is an unnecesary security risk, it is enough with mounting gtt configuration directory.

The correct way to run it would be like this, for example:
```

$ sudo docker run \
       --rm -it \
       -v ${PWD}/.local/share/.gtt/:/root/.local/share/.gtt \
       kriskbx/gitlab-time-tracker \
       report "ourorganization/aproject" --user=xxxx --from="2020-04-01" --to="2020-06-30"

```
I just tested this and it works, the current documentation does not.

**Comment by juanluisbaptiste** [comment]
Or better yet, if you specify the path to the config directory as specified above then you **_can_** use "~" to specify the path to your home directory (or ${HOME}):

```
$ sudo docker run \
       --rm -it \
       -v ~/.local/share/.gtt/:/root/.local/share/.gtt \
       kriskbx/gitlab-time-tracker \
       report "ourorganization/aproject" --user=xxxx --from="2020-04-01" --to="2020-06-30"

```

File: documentation.md (hunk 1 above)
Comment (cgdobre) [reply documentation.md:79]: We are not using an env vars here, but the output of the **pwd** command. Please note use of parentheses $**(**pwd**)**, not accolades ${pwd}

**Comment by cgdobre** [comment]
> Hi,
> 
> This PR does not fix the issue, first, env variable names are case sensitive and the correct env variable name for the current path value is `${PWD}`, `${pwd}` does return an empty value. Second, the volume mount for the /root directory using "~" is still present so the `docker run` command will still fail. Third, you don't need to mount the complete user home directory, that is an unnecesary security risk, it is enough with mounting gtt configuration directory.
> 
> The correct way to run it would be like this, for example:
> 
> ```
> 
> $ sudo docker run \
>        --rm -it \
>        -v ${PWD}/.local/share/.gtt/:/root/.local/share/.gtt \
>        kriskbx/gitlab-time-tracker \
>        report "ourorganization/aproject" --user=xxxx --from="2020-04-01" --to="2020-06-30"
> ```
> 
> I just tested this and it works, the current documentation does not.

Please use the comand line as mentioned in the doc. It does work. I have tested it. just make sure to use $(pwd), not ${pwd} (parentheses not accolades).

Both /root and /pwd volumes are necessary; one for **global** config and one for **local** config, which can partially or totally override the global config.

I haven't noticed any issues with using **~**. Can you tel me what error you are encountering?

**Comment by cgdobre** [comment]
> Or better yet, if you specify the path to the config directory as specified above then you **_can_** use "~" to specify the path to your home directory (or ${HOME}):
> 
> ```
> $ sudo docker run \
>        --rm -it \
>        -v ~/.local/share/.gtt/:/root/.local/share/.gtt \
>        kriskbx/gitlab-time-tracker \
>        report "ourorganization/aproject" --user=xxxx --from="2020-04-01" --to="2020-06-30"
> ```

If a global gtt config has not been created yet, then this complete path would cause an issue. This is why I would prefer to keep the home directory mapped into the container. You can change any of the /root and /pwd mounts in the script anyway if you wis to be more secure, but this more general script covers more use cases, probably trading off some security indeed.

**Comment by juanluisbaptiste** [comment]
> Please use the comand line as mentioned in the doc. It does work. I have tested it. just make sure to use $(pwd), not ${pwd} (parentheses not accolades).

Are you completely sure that pwd in lower case works for you ? for me it does not in any linux distro, tested on ubuntu, fedora, centos: 

```
$ echo ${pwd}

$
$ echo ${PWD}
/home/juancho
$

```
Shell env variables are case sensitive, all shell defined variables are uppercase according to the [IEE spec](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html). An excerpt from it:

> Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit. Other characters may be permitted by an implementation; applications shall tolerate the presence of such names. Uppercase and lowercase letters shall retain their unique identities and shall not be folded together. The name space of environment variable names containing lowercase letters is reserved for applications. Applications can define any environment variables with names from this name space without modifying the behavior of the standard utilities. 

And further down you can see the shell env variables which include PWD. I really don't understand why lowercase pwd works for you, in linux. 

> 
> Both /root and /pwd volumes are necessary; one for **global** config and one for **local** config, which can partially or totally override the global config.
> 

Could you elaborate on the use of a global config file when used inside a container ? 

> I haven't noticed any issues with using **~**. Can you tel me what error you are encountering?

One char paths like **~** (or **.**) are not allowed by docker as a volume path name [since a long time ago](https://github.com/docker/docker-ce/commit/1a062ddf1e0344e12153d1c549e5d2786757c004). There's also an [open issue](https://github.com/kriskbx/gitlab-time-tracker/issues/104) on this repo about this.

**Comment by juanluisbaptiste** [comment]
> If a global gtt config has not been created yet, then this complete path would cause an issue. This is why I would prefer to keep the home directory mapped into the container. You can change any of the /root and /pwd mounts in the script anyway if you wis to be more secure, but this more general script covers more use cases, probably trading off some security indeed.

Sorry it seems I do not understand the purpose of a _global_ config file that resides on a user home directory along with the _local_ override file. Normally, a global configuration file is installed system wide at a system path like _/etc_ so all users of the system can make use of it and then override it with a local config file on their home directory. So if both the local and global files are on the user's home directory, the global file is global relative to what ? why using only one config file is not enough ?

**Comment by klarkc** [comment]
@juanluisbaptiste You probabily not understood that the `$(pwd)` (which is not `{pwd}`) executes `pwd` in a sub-shell, `pwd` command (not variable) is part of GNU binutils, so it's already part of any linux variant.

About global/local. You might have a global config, overwritten by a local config. (Ex: you have a cross-project config inside your home directory to always query in Merge Requests, but for a given project (local directory, `pwd` output), you want to query only issues, so you can override that setting using the context aware file.

**Pull request merged**